geodata/models/imagery/base.py

```diff
@@ -0,0 +1,96 @@
+"""Base classes for raster dataset entries."""
+from django.contrib.gis.db import models
+from django.contrib.postgres import fields
+from django.db import transaction
+from django.db.models.signals import pre_save
+from django.dispatch import receiver
+from s3_file_field import S3FileField
+
+from .ifiles import ImageFile
+from ..common import ChecksumFile, ModifiableEntry, SpatialEntry
+from ..mixins import TaskEventMixin
+from ... import tasks
+
+
+class ImageEntry(ModifiableEntry):
+    """Single image entry, tracks the original file."""
+
+    name = models.CharField(max_length=100, blank=True, null=True)
+    description = models.TextField(null=True, blank=True)
+
+    instrumentation = models.CharField(
+        max_length=100,
+        null=True,
+        blank=True,
+        help_text='The instrumentation used to acquire these data.',
+    )
+
+    image_file = models.OneToOneField(ImageFile, null=True, on_delete=models.CASCADE)
+    driver = models.CharField(max_length=100)
+
+    # thumbnail = models.ImageField(blank=True, upload_to='thumbnails')
+
+    height = models.PositiveIntegerField()
+    width = models.PositiveIntegerField()
+    number_of_bands = models.PositiveIntegerField()
+    metadata = fields.JSONField(null=True)
+
+
+class ImageSet(ModifiableEntry):
+    """Container for many images."""
+
+    name = models.CharField(max_length=100, blank=True, null=True)
+    description = models.TextField(null=True, blank=True)
+
+    images = models.ManyToManyField(ImageEntry)
+
+
+class RasterEntry(SpatialEntry, ImageSet, TaskEventMixin):
```

Comment:
If we change the order of inheritance to `ImageSet, SpatialEntry, TaskEventMixin`, then name becomes the first column.  Currently, acquisition date is the first column.  Alternately, we could change the admin view to have name first.

Reply:
Ahh, I did not realize that happened. Good to learn how inheritance order affects the primary key/first column